addons/stock/models/stock_picking.py

```diff
@@ -1282,14 +1292,18 @@ def _autoconfirm_picking(self):
                 if picking.immediate_transfer:
                     picking.move_ids.write({'state': 'assigned'})
 
-    def _create_backorder(self):
+    def _create_backorder(self, split=False):
         """ This method is called when the user chose to create a backorder. It will create a new
         picking, the backorder, and move the stock.moves that are not `done` or `cancel` into it.
         """
         backorders = self.env['stock.picking']
         bo_to_assign = self.env['stock.picking']
         for picking in self:
-            moves_to_backorder = picking.move_ids.filtered(lambda x: x.state not in ('done', 'cancel'))
+            if split:
+                func = lambda x: not x.quantity_done and x.state not in ('done', 'cancel')
+            else:
+                func = lambda x: x.state not in ('done', 'cancel')
+            moves_to_backorder = picking.move_ids.filtered(func)
```

Comment:
Not bad, but since we are adding an argument a more useful one in the future would be a domain for the moves we want. I would add a `moves_domain`  argument instead of `split`. This will add versatility for any changes in the future. Also update the function description to explain the changes made
```suggestion
            if not moves_domain:
                moves_domain = "[('state', 'not in', ['done', 'cancel'])]"
            moves_to_backorder = picking.move_ids.filtered_domain(moves_domain)
```